apps/explorer/src/app/routes/types/block-explorer-response.d.ts

```diff
@@ -1,3 +1,5 @@
+import type { UnknownObject } from '../../components/nested-data-list';
+
 export interface BlockExplorerTransactionResult {
```

Comment:
suggestion (non-blocking): realised this is causing the necessity for the type casting in each component. the way you could potentially resolve this is by making this type a generic, by doing something like:

```
type TransactionResultCommand = 
    | ValidatorHeartbeat
    | SubmitOrder
    | StateVariableProposal
    | AmendLiquidityProvisionOrder
    | BatchMarketInstructions
    | ChainEvent;
    
export type BlockExplorerTransactionResult<T extends TransactionResultCommand> = {
  ...
  command: T
}
```

and the way you'd use it on the component props would be like:

```
const ComponentProps1 = {
   transaction: BlockExplorerTransactionResult<ValidatorHeartbeat>
} 

const ComponentProps2 = {
   transaction: BlockExplorerTransactionResult<SubmitOrder>
} 
```

Reply:
Thanks for teaching me to typescript better.

I continued the pattern that was there, but realise while doing it that really generating the types from the source would be better, so I've created https://github.com/vegaprotocol/frontend-monorepo/issues/2176. Thanks for telling me how to do this better regardless!